### PR TITLE
PA-180 Change email address twice in a row

### PR DESCRIPTION
**Pull request by benagain**

### CODE REVIEW

File: src/SFA.DAS.LoginService.Application/ChangeEmail/StartChangeEmail/StartChangeEmailRequest.cs [hunk 1]
```diff
@@ -1,11 +1,13 @@
 using MediatR;
 using System;
+using System.Security.Claims;
 
 namespace SFA.DAS.LoginService.Application.ChangeEmail.StartChangeEmail
 {
     public class StartChangeEmailRequest : IRequest<StartChangeEmailResponse>
     {
         public Guid ClientId { get; set; }
+        public string SubjectId { get; set; }
```
Comment (Pauljgraham) [inline src/SFA.DAS.LoginService.Application/ChangeEmail/StartChangeEmail/StartChangeEmailRequest.cs:10]: I can see the claim is called Sub and this is where we store the Email address, but do we need to call it SubjectId? And do we need the property

public string CurrentEmailAddress { get; set; }

Are they not the same thing?

File: src/SFA.DAS.LoginService.Web/Controllers/ChangeEmail/ConfirmChangeEmailController.cs [hunk 2]
```diff
@@ -44,6 +49,13 @@ public async Task<IActionResult> ConfirmChangeEmail(Guid clientId, [FromForm] Co
             {
                 SetModelState(response, model);
                 return View(model);
+}
+
```
Comment (Pauljgraham) [inline src/SFA.DAS.LoginService.Web/Controllers/ChangeEmail/ConfirmChangeEmailController.cs:53]: picky but layout looks wrong

File: src/SFA.DAS.LoginService.Web/Controllers/ChangeEmail/ConfirmChangeEmailController.cs [hunk 3]
```diff
@@ -44,6 +49,13 @@ public async Task<IActionResult> ConfirmChangeEmail(Guid clientId, [FromForm] Co
             {
                 SetModelState(response, model);
                 return View(model);
+}
+
+            var authenticated = await HttpContext.AuthenticateAsync();
+
+            if (authenticated.Succeeded)
+            {
+                await HttpContext.SignInAsync(authenticated.Principal, authenticated.Properties);
```
Comment (Pauljgraham) [inline src/SFA.DAS.LoginService.Web/Controllers/ChangeEmail/ConfirmChangeEmailController.cs:58]: Probably can't happen, but if authenticated.Suceeded return false. We still go to ChnageEmailSuccessful page

**Review by Pauljgraham — COMMENTED** [review]
Haven't tried it locally, just a few questions

File: src/SFA.DAS.LoginService.Application/ChangeEmail/StartChangeEmail/StartChangeEmailRequest.cs (hunk 1 above)
Comment (benagain) [reply src/SFA.DAS.LoginService.Application/ChangeEmail/StartChangeEmail/StartChangeEmailRequest.cs:10]: No, it's really `UserId`.  I've changed it 👍 

You're right about `CurrentEmailAddress` as well, it shouldn't be in the request.

File: src/SFA.DAS.LoginService.Web/Controllers/ChangeEmail/ConfirmChangeEmailController.cs (hunk 2 above)
Comment (benagain) [reply src/SFA.DAS.LoginService.Web/Controllers/ChangeEmail/ConfirmChangeEmailController.cs:53]: Fixed

File: src/SFA.DAS.LoginService.Web/Controllers/ChangeEmail/ConfirmChangeEmailController.cs (hunk 3 above)
Comment (benagain) [reply src/SFA.DAS.LoginService.Web/Controllers/ChangeEmail/ConfirmChangeEmailController.cs:58]: This addition isn't actually needed, I've removed it now.

**Review by Pauljgraham — APPROVED** [review]

**Pull request merged**